### DISCUSSION TITLE
How to create a predefined plan and can semantic kernel have more than one plan?

### DISCUSSION THREAD

**Discussion by Skyquek** [microsoft/semantic-kernel#6735] [Q&A] [answered]
In this page:
https://learn.microsoft.com/en-us/semantic-kernel/agents/planners/?tabs=Csharp

> Using predefined plans
There are likely common scenarios that your users will frequently ask for. To avoid the performance hit and the costs associated with planner, you can pre-create plans and serve them up to a user.
 This is similar to the front-end development adage coined by Aaron Swartz: "[Bake, don't fry](http://www.aaronsw.com/weblog/000404)." By pre-creating, or "baking," your plans, you can avoid generating them on the fly (i.e., "frying"). You won't be able to get rid of "frying" entirely when creating AI apps, but you can reduce your reliance on it so you can use healthier alternatives instead.
 To achieve this, you can generate plans for common scenarios offline, and store them as XML in your project. Based on the intent of the user, you can then serve the plan back up so it can be executed. By "baking" your plans, you also have the opportunity to create additional optimizations to improve speed or lower costs.

I am a bit confuse with `generate plans for common scenarios offline, and store them as XML in your project. Based on the intent of the user, you can then serve the plan back up so it can be executed.`. I think I need to use `Handlebar` for this, but I am curious on how I can use predefined plan? The tutorial https://github.com/microsoft/semantic-kernel/blob/main/dotnet/notebooks/05-using-the-planner.ipynb don't really show how we can create our own planner.

I am thinking of having an agent that can do Intent classification plugin and choose the right plan for the right task.

**Comment by sophialagerkranspandey** [comment, marked as answer]
Currently you can use handlebars for this instance. We are updating our documentation, and you should be able to learn more about planners in the new site that launched today.
